paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/CustomerAdapter.kt

```diff
@@ -0,0 +1,72 @@
+package com.stripe.android.paymentsheet.repositories
+
+import androidx.annotation.RestrictTo
+import com.stripe.android.ExperimentalSavedPaymentMethodsApi
+import com.stripe.android.model.PaymentMethod
+
+@ExperimentalSavedPaymentMethodsApi
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface CustomerAdapter {
+    /**
+     * Retrieves a list of payment methods attached to a customer
+     */
+    suspend fun fetchPaymentMethods(): Result<List<PaymentMethod>>
+
+    /**
+     * Adds a payment method to a customer
+     * @param paymentMethodId, the payment method to attach to the customer
+     */
+    suspend fun attachPaymentMethod(paymentMethodId: String)
+
+    /**
+     * Deletes the given payment method from the customer
+     * @param paymentMethodId, the payment method to detach from the customer
+     */
+    suspend fun detachPaymentMethod(paymentMethodId: String)
+
+    /**
+     * Set the selected payment method option.
+     * @param paymentOption, the [PersistablePaymentMethodOption] to save in the [DataStore]
+     */
+    suspend fun setSelectedPaymentMethodOption(paymentOption: PersistablePaymentMethodOption)
+
+    /**
+     * Fetch the persisted payment method option from the [DataStore]
+     */
+    suspend fun fetchSelectedPaymentMethodOption(): Result<PersistablePaymentMethodOption?>
+
+    /**
+     * Returns a client secret configured to the attach a new payment method to a customer.
+     * This will call your backend to retrieve a client secret if you have provided a
+     * setupIntentClientSecretProvider in the init call.
+     */
+    suspend fun setupIntentClientSecretForCustomerAttach(): Result<String>
+}
+
+/**
+ * A persistable payment method
+ */
+@ExperimentalSavedPaymentMethodsApi
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+sealed class PersistablePaymentMethodOption(open val id: String) {
```

Comment:
Why not use `SavedSelection`?

Reply:
`SavedSelection` is internal, we could open it up though!

Reply:
Oh, I guess if this will be exposed, we should keep this separate type. Should we name it `SavedPaymentMethod` then?

Reply:
This is the name that iOS is going with, I would like to keep it mostly consistent.